### DISCUSSION TITLE
What am I doing wrong?

### DISCUSSION THREAD

**Discussion by StoryHack** [neutralinojs/neutralinojs#856] [Q&A] [answered]
After I run "neu create" to start a project, I'm trying to test out some of neutralino's features. When I use "neu run" the app runs fine.

In the main.js, I added a line:

`await Neutralino.storage.setData('foo',"bar");`

Now when I try "neu run" on the project, I now get an error

`SyntaxError: Unexpected identifier 'Neutralino'`

What am I doing wrong?

**Comment by Blatman** [comment, marked as answer]
Can happen if the await line is not within an async function.

**Comment by pathange-s** [comment]
@StoryHack Try replacing this line with
```
async function test(){
    try{
        let data = await Neutralino.storage.setData('foo',"bar");
        console.log(`Data: ${data}`);
    } catch(e){
        console.log(e);
    }
}

test();
```
Also make sure you update the `nativeAllowList` with `storage.setData`

**Comment by phpsmarter** [comment]
If you use `async` and `await` , all you function call  need  async   keywords
so , neutralino init need async  wrap 
if you don't use `await`  keyword ,when  test function enter event loop , it can't find  neutralino object

```javascript 
async function test(){
    try{
        let data = await Neutralino.storage.setData('foo',"bar");
        console.log(`Data: ${data}`);
    } catch(e){
        console.log(e);
    }
}

async  function init(){
    await  Neutralino.init() 
    test()  // just waiting for init ,cause it is last calling don't need await keyword
}

init();

```

**Reply by pathange-s** [reply]
Yes, obviously !

**Comment by pegvin** [comment]
It is happening Because Top Level Awaits Are Currently Not Supported, So you'll have to Wrap it around a Async Fucntion Like This:

```javascript
async function setSomeData() {
  try {
    console.log((await Neutralino.storage.setData('foo',"bar")));
  } catch (e) {
    console.log(e);
  }
}

setSomeData();
```

Or You can call the function without the Await using `.then` and `.catch`

```javascript
Neutralino.storage.setData('foo',"bar")
  .then(function(response) {
    console.log(response);
  })
  .catch(function(err) {
    console.log(err);
  });

```
